### PR TITLE
[5.2] Refactor all instances of File to use framework

### PR DESCRIPTION
**Pull request by Hackwar**
### Summary of Changes
We are in the process of migrating away from the CMS filesystem package to use the framework version of it. This PR converts all remaining instances where File is used to use the framework version.

### Testing Instructions
Test that you can install extensions and that you can update Joomla.

OR

Codereview.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/components/com_joomlaupdate/src/Model/UpdateModel.php [hunk 1]
```diff
@@ -508,7 +508,11 @@ protected function downloadPackage($url, $target)
 
         // Make sure the target does not exist.
         if (is_file($target)) {
-            File::delete($target);
+            try {
+                File::delete($target);
+            } catch (FilesystemException $exception) {
+
+            }
```
Comment (HLeithner) [inline administrator/components/com_joomlaupdate/src/Model/UpdateModel.php]: We have logging 5 lines above, maybe log the exception?

File: administrator/components/com_installer/src/Model/InstallModel.php [hunk 2]
```diff
@@ -327,7 +328,10 @@ protected function _getPackageFromUpload()
         $tmp_src  = $userfile['tmp_name'];
 
         // Move uploaded file.
-        File::upload($tmp_src, $tmp_dest, false, true);
+        try {
+            File::upload($tmp_src, $tmp_dest, false, true);
+        } catch (FilesystemException $exception) {
```
Comment (bembelimen) [inline administrator/components/com_installer/src/Model/InstallModel.php:333]: When the upload fails, we should stop the execution...

File: administrator/components/com_joomlaupdate/src/Model/UpdateModel.php [hunk 3]
```diff
@@ -889,18 +895,21 @@ public function cleanUp()
...
+            if (is_file(JPATH_COMPONENT_ADMINISTRATOR . '/update.php')) {
+                File::delete(JPATH_COMPONENT_ADMINISTRATOR . '/update.php');
+            }
 
-        // Remove joomla.xml from the site's root.
-        if (is_file(JPATH_ROOT . '/joomla.xml')) {
-            File::delete(JPATH_ROOT . '/joomla.xml');
+            // Remove joomla.xml from the site's root.
+            if (is_file(JPATH_ROOT . '/joomla.xml')) {
+                File::delete(JPATH_ROOT . '/joomla.xml');
```
Comment (HLeithner) [inline administrator/components/com_joomlaupdate/src/Model/UpdateModel.php:910]: catch exception

File: administrator/components/com_joomlaupdate/src/Model/UpdateModel.php [hunk 4]
```diff
@@ -983,9 +994,9 @@ public function upload()
         $tmp_src  = $userfile['tmp_name'];
 
         // Move uploaded file.
-        $result = FileCMS::upload($tmp_src, $tmp_dest, false, true);
-
-        if (!$result) {
+        try {
+            File::upload($tmp_src, $tmp_dest, false);
+        } catch (FilesystemException $exception) {
             throw new \RuntimeException(Text::_('COM_INSTALLER_MSG_INSTALL_WARNINSTALLUPLOADERROR'), 500);
```
Comment (Fedik) [inline administrator/components/com_joomlaupdate/src/Model/UpdateModel.php]: have to chain the exceptions then, like:
```php
throw new \RuntimeException('Blabla error', 0, $exception);
```

The same for other places.

**Comment by Fedik** [comment]
@Hackwar a hint for debuging, hack the ErroDocument, and set status 200 https://github.com/joomla/joomla-cms/blob/e2170c8a6224ebb6ed754853de0d5e1b97ba1ce9/libraries/src/Document/ErrorDocument.php#L122

The thing that cypress always show blank page when the response code non 200.
This hack will force it to render joomla error page (unless it handled by Symfony error handler), and see them on screen shots in Artifacts.

**Comment by Hackwar** [comment]
The problem is, that we are surpressing the error page by having debug mode disabled. And in API there is no error page.

**Comment by Fedik** [comment]
> that we are surpressing the error page by having debug mode disabled

Yes, but you still will not see error on screenshot, because response non 200.

> And in API there is no error page.

hmhm, yea, that tricky

File: administrator/components/com_joomlaupdate/src/Model/UpdateModel.php (hunk 4 above)
Comment (Hackwar) [reply administrator/components/com_joomlaupdate/src/Model/UpdateModel.php]: Done

**Comment by degobbis** [comment]
I have tested this item :white_check_mark: successfully on 16a0a1e5ac9dff8cb71a914030005da2129c78ff<br /><br />Installing and uninstalling works as expected<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43362">issues.joomla.org/tracker/joomla-cms/43362</a>.</sub>

**Comment by exlemor** [comment]
I have tested this item :white_check_mark: successfully on 16a0a1e5ac9dff8cb71a914030005da2129c78ff<br /><br />I was able to install 4-5 simple to large components and confirmed they worked and uninstalled them successfully.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43362">issues.joomla.org/tracker/joomla-cms/43362</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43362">issues.joomla.org/tracker/joomla-cms/43362</a>.</sub>

**Comment by pe7er** [comment]
Thanks @Hackwar !

**Pull request merged**